### ISSUE TITLE
Devil may Cry 5 (601150) : Crashes or Locks up

### ISSUE THREAD

**Issue by shaumux** [Frogging-Family/wine-tkg-git#215]
- System: Gentoo
- Kernel: 5.9.7-zen
- GPU: GTX 1070
- Drivers: 455.38
- Crash Logs: [steam-601150.log](https://github.com/Frogging-Family/wine-tkg-git/files/5566983/steam-601150.log)
- Lockup  Logs: [steam-601150.log](https://github.com/Frogging-Family/wine-tkg-git/files/5566987/steam-601150.log)
- Build Configs: [proton-tkg.zip](https://github.com/Frogging-Family/wine-tkg-git/files/5566997/proton-tkg.zip)
- Built using docker arch

The game launches but crashes if I enter and exit the customize screen just before the missions or locks up after a minute or two of gameplay if I start the level, the sound and music seem to be still playing correctly, have tried with multiple builds.

**Comment by Tk-Glitch** [comment]
Looks like your GPU driver dies after hitting `err:nvapi:NvAPI_D3D11_SetDepthBoundsTest Failed to get wined3d device handle!`.
Try passing `PROTON_NVAPI_DISABLE=1 %command%` as launch option and see if the problem persists.

**Comment by shaumux** [comment]
Didn't help, still hung up but got an error this time
Logs: [steam-601150.log](https://github.com/Frogging-Family/wine-tkg-git/files/5567219/steam-601150.log)

**Comment by Tk-Glitch** [comment]
Ok now it's just the nvidia driver committing sudoku, so nvapi wasn't the actual cause. I'm noticing you're running the game @3440x1440, that might be an issue on a 1070 considering the huge video memory requirements. Can you please try with a lower resolution like 2560x1080 to see if it's any better? I can at least say this isn't a wine issue. This is either driver or DXVK related.

**Comment by shaumux** [comment]
Don't think its a memory issue, tired again, it just locked-up loading the level 
- Logs [steam-601150.log](https://github.com/Frogging-Family/wine-tkg-git/files/5567306/steam-601150.log)
I'll try at a lower resolution anyway.
I also tried with Proton-GE was able to complete the whole level, that is leading me to believe might be a DXVK issue here after your analysis

**Comment by Tk-Glitch** [comment]
You could also try DXVK's dxgi, by adding `PROTON_USE_WINE_DXGI=0` before `%command%` in your launch parameters.

**Comment by shaumux** [comment]
Same with DXVK's dxgi as well [Uploading steam-601150.log…]()
The seppuku continues.
I had removed `PROTON_NVAPI_DISABLE=1` while trying this though

**Comment by shaumux** [comment]
Tried with `PROTON_USE_WINE_DXGI=0` at 1920x1080 with the same results.
- Log [steam-601150.log](https://github.com/Frogging-Family/wine-tkg-git/files/5567666/steam-601150.log)
Also, this seems to be pretty random when it happens, sometimes it's a few minutes into gameplay, and at other times it's just after loading.

**Comment by Tk-Glitch** [comment]
And it takes Xorg with it :disappointed: I'll give the game a try on my end on intel/mesa to see if I can reproduce the issue. Have you tried with a different GPU driver yet? It might be worth giving 455.45.01 a try.

**Comment by shaumux** [comment]
Not sure what you mean by taking Xorg with it, since everything locks up, and no input works, I ssh into the system and issue a reboot command if thats useful

**Comment by Tk-Glitch** [comment]
So, I have tested the game with my intel broadwell iGPU and there are quite a few issues due to poor driver support. Flickering geometry on various objects/floors, notably. The game also refused to work with dx12 due to missing vulkan features and I had to force dx11. I was able to enjoy random (recoverable) hangs, but I'm not sure it's actually the same issue as I initially could finish the first mission a couple times, and even got to the 2nd one after the pretty long cutscenes before encountering my first hang.

Then I moved to a GTX 1050Ti equipped laptop that encountered *very* frequent (non recoverable) hangs either during loading screens or early on during gameplay. That looks similar to your experience. That was with 455 series drivers as well. Then I moved back to dx12 and it seemingly fixed it all. I was able to play through three missions without any issue. It might very well be an update that frogged it also, since the game would default to dx12 (on windows native) on pretty much any hardware able to run the game decently anyway. Along with no ingame toggle to select dx11, this is, to me, an admission of poor support for the dx11 renderer. I remember playing the game on Nvidia (980Ti) at launch on dx11 with no such issue so the first thing to try would be switching drivers. I'll try 450 series first, then if it's still unstable, give 440 a try.

For now I'll recommend you to switch to dx12 since your hardware can use vkd3d-proton. If you don't know how to do it, you just have to edit the `dmc5config.ini` file in the game dir to have `Capability=DirectX12` and `TargetPlatform=DirectX12`.

Regarding the customize screen, this is due to mfplat. An up to date Archlinux system with all gst-plugins and gst-libav packages installed coupled with a fresh proton-tkg 5.22 (or the "release" prebuilt available in this repo) will make it work.
